### ISSUE TITLE
CTCLoss reduction=’none‘ and calculate average in hand is not equal to 'mean'

### ISSUE THREAD

**Issue by carry-xz** [pytorch/pytorch#19838]
CTCLoss set reduction=’none‘ and calculate average by sum(loss)/len(loss) ,but the result was inconsistent with the result set reduction= 'mean'？

**Comment by cpuhrsch** [comment]
Hello @carry-xz, could I ask you for a script to reproduce this behavior? What environment are you working within (OS, PyTorch version etc.)? When opening an issue you should see the option to open up a bug report, which will have a list of all the information we typically need.

**Comment by fmassa** [comment]
@carry-xz this is expected, see the implementation of `CTCLoss` in https://github.com/pytorch/pytorch/blob/39b885cbbfc8c115069d49f5a6d27ea622bd05dc/aten/src/ATen/native/LossCTC.cpp#L366-L371

From [the documentation](https://pytorch.org/docs/stable/nn.html#torch.nn.CTCLoss)
> ‘mean’: the output losses will be divided by the target lengths and then the mean over the batch is taken
